### PR TITLE
icon thing

### PR DESCRIPTION
**Pull request by callumjt**
does that thing that #429 wants

### CODE REVIEW

**Comment by RedGuy12** [comment]
its aupposed to be modules/settings.ts

**Comment by RedGuy12** [comment]
also "resolves #429"

**Comment by callumjt** [comment]
> its aupposed to be modules/settings.ts

thats where i put it in

**Comment by callumjt** [comment]
> > its aupposed to be modules/settings.ts
> 
> thats where i put it in

oh wait, i uploaded to wrong place

File: modules/settings.ts [hunk 1]
```diff
@@ -214,9 +216,13 @@ export async function mentionUser(
 	interactor: { id: Snowflake },
 	guild = config.guild,
 ) {
-	const { useMentions } = getSettings(interactor);
 	const id = user instanceof User ? user.id : user;
-	if (useMentions) return userMention(id);
+
+	const member = (await guild.members.fetch(id).catch(() => void 0)) as GuildMember;
```
Comment (RedGuy12) [inline modules/settings.ts:221]: don't cast :sob: just use optional chaining???

File: modules/settings.ts [hunk 2]
```diff
@@ -225,5 +231,7 @@ export async function mentionUser(
 
 	const { displayName } =
 		user instanceof User ? user : (await client.users.fetch(user).catch(() => void 0)) ?? {};
-	return displayName ? hyperlink(censor(displayName), url) : userMention(id);
+	return displayName
+		? `${hyperlink(censor(displayName), url)} ${icon} `
+		: `${userMention(id)} ${icon} `;
```
Comment (RedGuy12) [inline modules/settings.ts:236]: there should not be a trailing space

File: modules/settings.ts [hunk 3]
```diff
@@ -214,9 +216,13 @@ export async function mentionUser(
...
 ) {
-	const { useMentions } = getSettings(interactor);
 	const id = user instanceof User ? user.id : user;
-	if (useMentions) return userMention(id);
+
+	const member = (await guild.members.fetch(id).catch(() => void 0)) as GuildMember;
+	const icon = member.roles.icon?.unicodeEmoji;
+
+	const { useMentions } = getSettings(interactor);
+	if (useMentions) return icon + userMention(id);
```
Comment (RedGuy12) [inline modules/settings.ts:225]: the ping should go first, then a space, then the emoji

**Review by RedGuy12 — CHANGES_REQUESTED** [review]

File: modules/settings.ts (hunk 1 above)
Comment (callumjt) [reply modules/settings.ts:221]: Wha

File: modules/settings.ts (hunk 2 above)
Comment (callumjt) [reply modules/settings.ts:236]: Uhh how did that get in there

File: modules/settings.ts (hunk 1 above)
Comment (RedGuy12) [reply modules/settings.ts:221]: image
image

**Comment by callumjt** [comment]
@RedGuy12 I'm assuming this won't be merged until you find a way to handle custom icons?

**Comment by callumjt** [comment]
Could just detect if it's not Unicode emoji and if it isn't display mod badge instead since only mods have custom icons?

**Comment by RedGuy12** [comment]
that's not a correct statement

**Comment by callumjt** [comment]
> that's not a correct statement

Elaborate?

**Comment by RedGuy12** [comment]
> > that's not a correct statement
> 
> Elaborate?

for example: active booster, sa dev

**Comment by callumjt** [comment]
> > > that's not a correct statement
> > 
> > 
> > Elaborate?
> 
> for example: active booster, sa dev

I thought boosters don’t get icons?

**Comment by RedGuy12** [comment]
those two roles, among others, have role icons

**Comment by RedGuy12** [comment]
those two roles, among others, have role icons

**Comment by RedGuy12** [comment]
@dependabot rebase

**Pull request closed without merging**